packages/jest-runtime/src/index.ts

```diff
@@ -302,6 +320,83 @@ class Runtime {
     return cliOptions;
   }
 
+  // unstable as it should be replaced by https://github.com/nodejs/modules/issues/393, and we don't want people to use it
+  unstable_shouldLoadAsEsm = Resolver.unstable_shouldLoadAsEsm;
+
+  private async loadEsmModule(
+    modulePath: Config.Path,
+    query = '',
+  ): Promise<VMModule> {
+    const cacheKey = modulePath + query;
+
+    if (!this._esmoduleRegistry.has(cacheKey)) {
+      const context = this._environment.getVmContext!();
+
+      invariant(context);
+
+      if (this._resolver.isCoreModule(modulePath)) {
+        const core = await this._importCoreModule(modulePath, context);
+        this._esmoduleRegistry.set(cacheKey, core);
+        return core;
+      }
+
+      const transformedFile = this.transformFile(modulePath, {
+        isInternalModule: false,
+        supportsDynamicImport: true,
+        supportsStaticESM: true,
+      });
+
+      const module = new SourceTextModule(transformedFile.code, {
+        context,
+        identifier: modulePath,
+        importModuleDynamically: (
+          specifier: string,
+          referencingModule: VMModule,
+        ) =>
+          this.loadEsmModule(
+            this._resolveModule(referencingModule.identifier, specifier),
+          ),
+        initializeImportMeta(meta: JestImportMeta) {
+          meta.url = pathToFileURL(modulePath).href;
+          // @ts-ignore TODO: fill this
+          meta.jest = {};
```

Comment:
Do we want to provide Jest in this way at all, or do we want to rely on `import from '@jest/globals'` for ESM instead?

Reply:
Not sure. The `@jest/globals` thing is a thing I came of with this morning. I think I like importing more than adding to `meta` (it's easier to type which usually means it's easier to consume as ell), so maybe just drop it?